app/controllers/SurveyController.scala

```diff
@@ -0,0 +1,121 @@
+        val numMissionsCompleted: Int = MissionTable.countCompletedMissionsByUserId(UUID.fromString(userId))
+
+        val allSurveyQuestions = SurveyQuestionTable.listAll
+        val allSurveyQuestionIds = allSurveyQuestions.map(_.surveyQuestionId)
+        val answeredQuestions = submission.answeredQuestions
+        val answeredQuestionIds = submission.answeredQuestions.map(_.surveyQuestionId.toInt)
+        val unansweredQuestionIds = allSurveyQuestionIds diff answeredQuestionIds
+        // Iterate over all the questions and check if there is a submission attribute matching question id.
+        // Add the associated submission to the user_submission tables for that question
+
+
+        answeredQuestions.foreach{ q =>
+          val questionId = q.surveyQuestionId.toInt
+          val temp_question = SurveyQuestionTable.getQuestionById(questionId)
+          temp_question match{
+            case Some(question) =>
+              if (question.surveyInputType != "free-text-feedback") {
+                val userSurveyOptionSubmission = UserSurveyOptionSubmission(0, userId, question.surveyQuestionId, Some(q.answerText.toInt), timestamp, numMissionsCompleted)
+                val userSurveyOptionSubmissionId: Int = UserSurveyOptionSubmissionTable.save(userSurveyOptionSubmission)
+              }
+              else {
+                val userSurveyTextSubmission = UserSurveyTextSubmission(0, userId, question.surveyQuestionId, Some(q.answerText), timestamp, numMissionsCompleted)
+                val userSurveyTextSubmissionId: Int = UserSurveyTextSubmissionTable.save(userSurveyTextSubmission)
+              }
+            case None =>
+              None
+          }
+        }
+        unansweredQuestionIds.foreach{ questionId =>
+          val temp_question = SurveyQuestionTable.getQuestionById(questionId)
+          temp_question match{
+            case Some(question)=>
+              if(question.surveyInputType != "free-text-feedback"){
+                val userSurveyOptionSubmission = UserSurveyOptionSubmission(0, userId, question.surveyQuestionId, None, timestamp, numMissionsCompleted)
+                val userSurveyOptionSubmissionId: Int = UserSurveyOptionSubmissionTable.save(userSurveyOptionSubmission)
+              }
+              else{
+                val userSurveyTextSubmission = UserSurveyTextSubmission(0, userId, question.surveyQuestionId, None, timestamp, numMissionsCompleted)
+                val userSurveyTextSubmissionId: Int = UserSurveyTextSubmissionTable.save(userSurveyTextSubmission)
+              }
+            case None =>
+              None
+          }
+        }
+
+        Future.successful(Ok(Json.obj("survey_success" -> "True")))
+      }
+    )
+
+  }
+
+  def shouldDisplaySurvey = UserAwareAction.async { implicit request =>
+    val userId: UUID = request.identity match {
+      case Some(user) => user.userId
+      case None =>
+        val user: Option[DBUser] = UserTable.find("anonymous")
```

Comment:
How do you prevent it from showing it to anon users?

Reply:
If you've noticed, the survey html isnt generated for the audit page in the case of anon users. Same for the case of the survey javascript